### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by eyip002**
Added board peripherals, route orientation, and reversed blocks

### CODE REVIEW

**Review by trinnguyen — COMMENTED** [review]
@eyip002 I put some feedbacks to address a minor bug and suggestions to change. I added some more unit tests to cover the issue, check out the test result in this pull request to get the counter example.

In general, above are just some small things. The grammar with YAML exporter are already implemented properly and good to go.

File: src/de.uniba.swt.dsl/src/de/uniba/swt/dsl/common/generator/yaml/exports/ExtraBlockElementYamlExporter.java [hunk 1]
```diff
@@ -63,6 +63,7 @@ protected String getId(ExtraBlockElement element) {
         list.add(Tuple.of("length", String.format("%.2f%s", length, unit.getLiteral().toLowerCase())));
         list.add(Tuple.of("main", mainSegs));
         list.add(Tuple.of("overlaps", overlaps));
+        list.add(Tuple.of("reversed", element.getBlockElement().isReversed()));
```
[the remark this answers is not in the document]
Comment (eyip002) [reply src/de.uniba.swt.dsl/src/de/uniba/swt/dsl/common/generator/yaml/exports/ExtraBlockElementYamlExporter.java:66]: Done

File: src/de.uniba.swt.dsl/resources/standardlib.bahn [hunk 2]
```diff
@@ -109,21 +114,21 @@ signaltypes
...
 end
 
 # Board peripheral are set directly to the
 # ports, which are active high
 peripheraltypes
     onebit
         aspects
-            on 0x01
-            off 0x00
+            high 0x01
```
[the remark this answers is not in the document]
Comment (eyip002) [reply src/de.uniba.swt.dsl/resources/standardlib.bahn:128]: Think of creating TrackStateType in standardlib.bahn similar to ConfigKey.

File: src/de.uniba.swt.dsl/src/de/uniba/swt/dsl/validation/BahnValidator.java [hunk 3]
```diff
@@ -154,7 +154,7 @@ public void validatePeripheralsProperty(PeripheralsProperty prop) {
 
             return null;
         }, BahnPackage.Literals.SIGNALS_PROPERTY__ITEMS);
-        validateUniqueName(prop.getItems(), SignalElement::getName, BahnPackage.Literals.PERIPHERALS_PROPERTY__ITEMS);
+        validateUniqueName(prop.getItems(), PeripheralElement::getName, BahnPackage.Literals.PERIPHERALS_PROPERTY__ITEMS);
```
[the remark this answers is not in the document]
Comment (trinnguyen) [reply src/de.uniba.swt.dsl/src/de/uniba/swt/dsl/validation/BahnValidator.java:168]: Solved now!

**Pull request merged**